mautic/mautic
# Fix focus campaign action - add local cache to tracking helper

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR prevent waiting for update session If campaign action is executed after page visits. 
Tracking helper in first step try check local cache, then check in session.




[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Create campaign with Page visits decision and after Focus action

![image](https://user-images.githubusercontent.com/462477/46875968-65bcd080-ce3d-11e8-8bd1-d500b4741223.png)

2. The action is not executed every time. It's little tricky test it, but If I run this campaign on production I have few Focus fired, even all action are executed. Just check focus stats and it's not same (focus stats are lower) like campaign action triggered stats. 

#### Steps to test this PR:
1.  Repeat same camapign
2.  Focus should fire every time


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/6723"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
